numpy/core/arrayprint.py

```diff
@@ -706,6 +706,34 @@ def _extendLine(s, line, word, line_width, next_line_prefix, legacy):
     return s, line
 
 
+def _extendLine_pretty(s, line, word, line_width, next_line_prefix, legacy):
+    """
+    Extends line with nicely formatted (possibly multi-line) string ``word``.
+    """
+    words = word.splitlines()
+    if len(words) == 1 or legacy == '1.13':
+        return _extendLine(s, line, word, line_width, next_line_prefix, legacy)
+
+    line_length = len(line)
+    max_word_length = max(len(word) for word in words)
+    if line_length + max_word_length > line_width and \
+            len(line) > len(next_line_prefix):
+        s += line.rstrip() + '\n'
+        line = next_line_prefix + words[0]
+        indent = next_line_prefix
+    else:
+        line += words[0]
+        indent = line_length*' '
```

Comment:
```suggestion
    max_word_length = max(len(word) for word in words)
    if len(line) + max_word_length > line_width and \
            len(line) > len(next_line_prefix):
        s += line.rstrip() + '\n'
        line = next_line_prefix + words[0]
        indent = next_line_prefix
    else:
        indent = ' '*len(line)
        line += words[0]
```
It looks like you can eliminate the need for defining the extra `line_length` variable.